lib/ex_unit/lib/ex_unit/diff.ex

```diff
@@ -286,7 +286,7 @@ defmodule ExUnit.Diff do
   defp diff_pin({:^, _, [var]} = pin, right, %{pins: pins} = env) do
     identifier = var_context(var)
     %{^identifier => pin_value} = pins
-    {diff, post_env} = diff_value(pin_value, right, env)
+    {diff, post_env} = diff_value(Macro.escape(pin_value), right, env)
 
     diff_left = update_diff_meta(pin, not diff.equivalent?)
     {%{diff | left: diff_left}, post_env}
```

Comment:
```suggestion
    {diff, post_env} = diff_value(pin_value, right, %{env | context: :===})

    diff_left = update_diff_meta(pin, not diff.equivalent?)
    {%{diff | left: diff_left}, %{post_env | context: :match}}
```

Reply:
diff_value works on values, so we shouldn't pass Macro.escape (an AST), the correct fix is to tell we should use a different context. :)